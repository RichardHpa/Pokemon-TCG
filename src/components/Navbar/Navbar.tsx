import { ThemeToggler } from 'components/ThemeToggler'
import { NavLink } from 'components/NavLink'
import { Link } from 'react-router-dom'

export const Navbar = () => {
  return (
    <nav className='border-gray-200 px-2 sm:px-4 py-2.5'>
      <div className='container flex flex-wrap justify-between items-center mx-auto border-b py-2'>
        <div className='flex'>
          <Link
            className='mr-8 flex p-0 text-2xl tracking-tighter no-underline text-gray-700 dark:text-white'
            to='/'
          >
            <span className='inline-block font-medium'>Pokemon Card Checker</span>
          </Link>
          <NavLink to='/'>Home</NavLink>
          <NavLink to='/sets'>Sets</NavLink>
          <NavLink to='/search'>Search</NavLink>
        </div>
        <ThemeToggler />
      </div>
    </nav>
  )
}
