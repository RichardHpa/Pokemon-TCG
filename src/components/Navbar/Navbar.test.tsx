import { render, screen } from 'tests/renderHelpers'

import { Navbar } from './Navbar'

describe('Navbar', () => {
  test('renders Navbar', () => {
    render(<Navbar />)

    const link = screen.getByRole('link', { name: /Pokemon Card Checker/i })

    expect(link).toBeInTheDocument()
  })

  test('renders all links', () => {
    render(<Navbar />)

    expect(screen.getAllByRole('link')).toHaveLength(4)

    const homeLink = screen.getByRole('link', { name: /home/i })
    const setsLink = screen.getByRole('link', { name: /sets/i })
    const searchLink = screen.getByRole('link', { name: /search/i })

    expect(homeLink).toBeInTheDocument()
    expect(setsLink).toBeInTheDocument()
    expect(searchLink).toBeInTheDocument()
  })

  test('renders Navbar snapshot', () => {
    const { container } = render(<Navbar />)
    expect(container).toMatchSnapshot()
  })
})
