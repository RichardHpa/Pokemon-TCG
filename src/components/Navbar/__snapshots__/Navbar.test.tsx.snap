// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders Navbar snapshot 1`] = `
<div>
  <nav
    class="border-gray-200 px-2 sm:px-4 py-2.5"
  >
    <div
      class="container flex flex-wrap justify-between items-center mx-auto border-b py-2"
    >
      <div
        class="flex"
      >
        <a
          class="mr-8 flex p-0 text-2xl tracking-tighter no-underline text-gray-700 dark:text-white"
          href="/"
        >
          <span
            class="inline-block font-medium"
          >
            Pokemon Card Checker
          </span>
        </a>
        <a
          aria-current="page"
          class="text-gray-700 dark:text-gray-200 hover:text-gray-900 dark:hover:text-gray-100 px-3 py-2 rounded-md text-sm font-medium hover:underline active"
          href="/"
        >
          Home
        </a>
        <a
          class="text-gray-700 dark:text-gray-200 hover:text-gray-900 dark:hover:text-gray-100 px-3 py-2 rounded-md text-sm font-medium hover:underline"
          href="/sets"
        >
          Sets
        </a>
        <a
          class="text-gray-700 dark:text-gray-200 hover:text-gray-900 dark:hover:text-gray-100 px-3 py-2 rounded-md text-sm font-medium hover:underline"
          href="/search"
        >
          Search
        </a>
      </div>
      <button
        aria-label="Switch to dark mode"
        class="inline-flex align-middle justify-center rounded-full p-2 text-inherit-500 hover:bg-inherit-500 hover:text-inherit-500 border-transparent hover:bg-opacity-25"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6 inline-block"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9.528 1.718a.75.75 0 01.162.819A8.97 8.97 0 009 6a9 9 0 009 9 8.97 8.97 0 003.463-.69.75.75 0 01.981.98 10.503 10.503 0 01-9.694 6.46c-5.799 0-10.5-4.701-10.5-10.5 0-4.368 2.667-8.112 6.46-9.694a.75.75 0 01.818.162z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </nav>
</div>
`;
